li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/utils/PrimitiveEncoderDecoder.java

```diff
@@ -120,6 +103,32 @@ public static int decodeInt(byte[] input, int pos) {
    * @return a decoded long
    */
   public static long decodeLong(byte[] input, int pos) {
+    sanityCheck(input, pos, LONG_SIZE);
```

Comment:
Not sure I follow the changes in this class. Can you please explain?  Is it just refactoring code ?

Reply:
Yes, it is just refactoring. In the first version, I add a new encode/decode method for boolean type for adding header. When I was doing that, I found there were some duplicate codes for sanity check. So I extracted them. After removing extra header, I also removed unused encodeBoolean method but left the sanity check method.

Reply:
Ok